cred_helpers.go

```diff
@@ -42,19 +48,35 @@ func SetupCredHelpers(refs ...string) error {
 				continue
 			}
 			credHelpers[registry] = ch.helper
-			added = true
 		}
 	}
-	if !added {
+
+	if len(credHelpers) == 0 {
 		return nil
 	}
+
+	for k, v := range credHelpers {
+		ch, ok := config["credHelpers"].(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("failed to parse docker config 'credHelpers'")
+		}
+
+		if _, ok := ch[k]; ok {
+			continue
+		}
+
+		ch[k] = v
```

Comment:
Suggestion: swap `ok` -> `!ok` and move this line into the loop above in place of `continue`.